### DISCUSSION TITLE
Импорт таблицы циклов вместе с расчетными таблицами

### DISCUSSION THREAD

**Discussion by nshy** [totumonline/totum-ru-issues-and-discussions#219] [Как это сделать?] [answered]
Привет!

Примеряю Totum к своей задачке. Задачка включает в себя бронирование мест в комнатах общежития. Подумал что можно было бы сделать комнаты общежития таблицей циклов, а места в комнате - расчетной таблицей. Проблема в экспорте\импорте таблиц в Totum в\из csv. Я могу эскпортировать таблицу циклов и каждую расчетную таблицу отдельно, а хотелось бы выгрузить одним файлом таблицу циклов и все связанные расчетные таблицы. В обратную сторону видимо так же, видимо не смогу загрузить из одного файла.

**Comment by erpic-ru** [comment, marked as answer]
Привет!

Пара моментов (на мой взгляд):

1. В циклах имеет смысл размещать документы, действия и расчеты. Например, бронирование или сам договор аренды. А места в комнатах больше похожи на простую таблицу с деревом по комнатам, с которыми уже будут происходить какие-то события в циклах
2. Можно сделать обработчик для загрузки данных и какую-то временную таблицу для выстаскивания данных из циклов. Но зачем настраивать двухсторонний обмен в CSV?  Следует всеми силами этого избегать

**Reply by nshy** [reply]
Не везде возможны интеграции. Общежитие предоставляет список комнат для проведения ивента. Перед ивентом им надо скинуть список кто в какой комнате живет.

**Reply by ttm-alexeypolunin** [reply]
Нет никакой выгрузки таблицы циклов и данных внутри циклов одновременно. Что бы я сделал для этой задачи — я бы сформировал временную таблицу в которую вытащил из циклов всю необходимую информацию — она и выгружается в .csv

Но мне кажется, что здесь глобальная ошибка — цикл это эвент. Так как повторяющееся событие это эвент. Внутри эвента у вас рассадка. Я бы сделал строками комнаты, колонками места в комнате.

Если у вас есть в .xlsx образец — скинте, нам будет понятнее.

**Reply by nshy** [reply]
С выгрузкой понятно. Насчет ивент это цикл. Ивенты редкие, можно считать раз в год, так что процесс строится для одного ивента. Но не буду вдаваться в подробности. Хорошая идея насчет колонок для мест в комнате.

Спасибо за totum, надеюсь удастся его применить для решения в нашем случае.

Очень хочется увидеть в мастерклассах решение для более-менее реальной задачи, например той что во втором ролике. Даже не столько в плане как его закодить, а как работа с ним выглядит с точки зрения пользователей. По каждой роли, и какие между ними есть коммуникации за пределами тотума.

**Reply by ttm-alexeypolunin** [reply]
Сейчас выпустим PRO, я запишу один мастеркласс для микрорешения на 3 ролика (это будет реальное решение для отдела большой компании).

> Проблема с мастерклассами такая, что большими их не сделаешь (смотреть не будут), а в маленький нифига не впихнуть. 

> Возможно мы сделаем какие-нибудь вебинары/стримы/хз-что, на которых я массово буду на вопросы отвечать. Может 1 раз в месяц такое попробуем делать.

А так можно взять у нас [платную консультацию](https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/20#discussioncomment-3535961) или [поддержку](https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/20#discussioncomment-3584872).
